### DISCUSSION TITLE
Option - Only seed a torrent if seeders are less than X

### DISCUSSION THREAD

**Discussion by ageekhere** [transmission/transmission#5745] [Ideas]
Add an option to only seed a torrent if the number of seeders for that torrent are less then X.

**Comment by valcomm** [comment]
> Add an option to only seed a torrent if the number of seeders for that torrent are less then X.

What other bittorrent client provides such functionality?
Could you also define 'number of seeders' please?

**Comment by ageekhere** [comment]
"What other bittorrent client provides such functionality" i do not think any offer this.
"Could you also define 'number of seeders' please?"

Ok lets give an example, lets say you are seeding 3 torrents.
Torrent 1 has a total of 10 seeders 100 leechers
Torrent 2 has a total of 100 seeders 100 leechers
Torrent 3 has a total of 0 seeders 50 leechers
So if I set the option to only seed a torrent if the number of seeders within the torrent are less than 5 then the outcome would be
Torrent 3 uploads
Torrent 2 paused uploading
Torrent 1 paused uploading

So if you have only a limited amount of upload speed but do not want a torrent to runout of seeders you can set it to only seed when needed.

**Comment by dantearmok** [comment]
Not sure I like this option.  A torrent's lifeblood is seeders - the more seeders, the better a torrent's "health".  And the more seeders, the less load is presented to each peer.

**Comment by ageekhere** [comment]
Maybe prioritize torrents with low seeder count would be a better option.

**Comment by dantearmok** [comment]
Transmission already has a priority setting.  Try using that for a while, to see if it has any long term viability for you.

**Comment by Pentaphon** [comment]
> Add an option to only seed a torrent if the number of seeders for that torrent are less then X.

This is ridiculous. All clients should always be seeding.

**Comment by ageekhere** [comment]
Well what some users do atm is when a torrent is finished they seed for a bit and if they see that the torrent has a healthy number of seeders they stop seeding the torrent to focus their bandwidth else where. The idea was for these users is to provide an option to have the torrent be paused and only resume seeding if the torrent starts to run out of seeders.

Now users can do this manually, have the torrent paused and check now and then on the health of the torrent. What I suggested was to automate this.

Example, say someone is seeding 100 torrents, instead of their bandwidth being divided among 100 torrents (say most being very healthy) 
they can focus their bandwidth on new uploaded torrents, so they just pause the old torrents to focus on the new uploads. 
The idea was that the paused torrents could auto unpause if a torrent's health drops.

**Comment by Pentaphon** [comment]
> Well what some users do atm is when a torrent is finished they seed for a bit and if they see that the torrent has a healthy number of seeders they stop seeding the torrent to focus their bandwidth else where. The idea was for these users is to provide an option to have the torrent be paused and only resume seeding if the torrent starts to run out of seeders.

I have never heard of somebody doing this and I have never head of a client that auto-senses peers like this. 

What people actually do is limit their upload slots, set up queuing and just seed as they normally would. What you are asking for is beyond the scope of a torrent client and something that will probably never be gotten to because this project has barely enough devs to keep it caught up with the latest coding standards and catching up with modern features that other clients have, much less creating a whole new feature that no client has ever had. I suggest just learning how to keep your client optimized for your connection.

**Comment by ageekhere** [comment]
@Pentaphon Thanks i have look into queuing.
